test/unit/test_unit_tests.py

```diff
@@ -536,29 +536,28 @@ def test_augment_on_empty_test_suite(self) -> None:
             config['general']['reports_path'] = app_name + '-reports_dir'
         shutil.rmtree(config['general']['test_directory'], ignore_errors=True)
         os.makedirs(config['general']['test_directory'])
-        monolithic_dir = os.path.join(config['general']['test_directory'], 'monolithic')
-        build_file = build_util.generate_build_xml(
-            app_name=app_name,
-            build_type='gradle',
-            monolith_app_path=config['general']['monolith_app_path'],
-            app_classpath=build_util.get_build_classpath(config),
-            test_root_dir=config['general']['test_directory'],
-            test_dirs=[monolithic_dir],
-            # partitions_file=None,
-            target_class_list=[],
-            main_reports_dir=config['general']['reports_path'],
-            app_packages=[],  # for coverage-based augmentation
-            collect_codecoverage=True,  # for coverage-based augmentation
-            offline_instrumentation=True,
-            output_dir=output_dir
-        )
         config['general']['build_type'] = 'gradle'
         test_files = [None, None]
         for use_for_augmentation in [False, True]:
             config['dev_tests']['use_for_augmentation'] = use_for_augmentation
-            shutil.rmtree(monolithic_dir, ignore_errors=True)
-            os.makedirs(monolithic_dir)
+            shutil.rmtree(config['general']['test_directory'], ignore_errors=True)
```

Comment:
@haim-kermany we are not allowed to delete the test directory - it may contain user stuff in addition to the tests we put there

Reply:
it is the testing code...

Reply:
please notice that the cli always assumed that the user is supplying an empty directory.
1. in generate, we always delete it before copying the <test_dir>_tmp to it.
2. in augmentation, we refer to the content of it.
3. in creating a build file, we add all the content of in (beside build anf target)

Reply:
it is deleted at generate.py line 181. right before coping the tst_dir_tmp.
we deleted it at the old code, and the new code.
fixing the deletion is easy. but there should be also a fix in augmentation, and build file creation.
and this is not trivial (both augmentation and build_utils assumes that all the tests from the tests directory was created by tkltest)
we should give it another thought. but I think it should be done on a separate issue.